### ISSUE TITLE
List literals can have mismatched types due to an off-by-one error

### ISSUE THREAD

**Issue by smarx** [ethereum/vyper#1692]
This check should use `> 0` instead of `> 1`:

https://github.com/ethereum/vyper/blob/c296b2d7532d913103aad494b749f8179a3acddc/vyper/parser/expr.py#L1066-L1067

The result is inconsistent type checking for list literals:

```python
@public
def test():
    x: uint256[3] = [2**255, 1, 3] # Compiles without error
    y: uint256[3] = [1, 2**255, 3] # Error: Lists may only contain one type
```

**Comment by charles-cooper** [comment]
should be fixed in https://github.com/ethereum/vyper/pull/1673

**Comment by charles-cooper** [comment]
confirmed fixed as of f7211c130a180156e64f7d6d141db30ef02e0c14
```
$ vyper -f ir /dev/stdin <<EOF

@public
def test():
    x: uint256[3] = [2**255, 1, 3] # Compiles without error
    y: uint256[3] = [1, 2**255, 3] # Error: Lists may only contain one type
EOF

Error compiling: /dev/stdin
vyper.exceptions.TypeMismatchException: line 4:20 Lists may only contain one type
     3 def test():
---> 4     x: uint256[3] = [2**255, 1, 3] # Compiles without error
---------------------------^
     5     y: uint256[3] = [1, 2**255, 3] # Error: Lists may only contain one type
```
